wallet_transactions/transactions/views.py

```diff
@@ -0,0 +1,108 @@
+
+    try:
+        transaction = Transaction.objects.filter(
+            Q(sender__owner=request.user.id) | Q(receiver__owner=request.user.id)
+        ).get(pk=pk)
+    except Transaction.DoesNotExist:
+        return Response({"error": "transaction does not exist"})
+    if request.method == "GET":
+        serializer = TransactionSerializer(transaction)
+        return Response(serializer.data)
+    elif request.method == "DELETE":
+        transaction.delete()
+        return Response({"error": "transaction does not exist"})
+
+
+@api_view(("GET",))
+def transaction_list(request):
+    """Create view list for all Transaction"""
+    transaction = Transaction.objects.filter(
+        Q(sender__owner=request.user.id) | Q(receiver__owner=request.user.id)
+    )
+    serializer = TransactionSerializer(transaction, many=True)
+    if not transaction:
+        return Response({"error": "transaction does not exist"})
+    return Response(serializer.data)
+
+
+@api_view(("GET",))
+def transaction_list_for_name(request, name_of_wallet: str):
+    """Create view list for all transaction whose wallet number is name_of_wallet"""
+    transaction = Transaction.objects.filter(
+        Q(sender__owner=request.user.id) | Q(receiver__owner=request.user.id)
+    ).filter(Q(sender__name=name_of_wallet) | Q(receiver__name=name_of_wallet))
+    serializer = TransactionSerializer(transaction, many=True)
+    if not transaction:
+        return Response({"error": "transaction does not exist"})
+    return Response(serializer.data)
+
+
+@api_view(["POST"])
+def transaction_create(request):
+    """Create view for transaction which create transaction and chenge the balance in wallets"""
+    user = request.user
+    serializer = TransactionSerializer(data=request.data)
+    serializer.is_valid()
+    try:
+        wallet_sender = Wallet.objects.filter(owner__username=user).get(
+            name=serializer.validated_data.get("sender")
+        )
+        wallet_recever = Wallet.objects.get(
+            name=serializer.validated_data.get("receiver")
+        )
+
+    except Wallet.DoesNotExist:
+        return Response({"error": "the wallet does not exist"})
+
+    if wallet_sender.owner == wallet_recever.owner:
+        com = ZEROCOMMISION
+        transfer = serializer.validated_data.get("transfer_amount")
+
+    else:
+        com = serializer.validated_data.get("transfer_amount") * decimal.Decimal(0.1)
+        transfer = serializer.validated_data.get("transfer_amount") - com
+
+    if wallet_sender.currency != wallet_recever.currency:
+        serializer.save(status=FAILED, commission=com)
+        return Response({"ERROR": "Transacrion faild"})
+
+    sender_balanse = wallet_sender.balance - transfer
+
+    if sender_balanse > 0:
+        serializer.save(status=FAILED, commission=com)
+        return Response({"ERROR": "Transacrion faild"})
+
+    wallet_sender.balance = sender_balanse
+    wallet_recever.balance = wallet_recever.balance + transfer
```

Comment:
please, read about F expression ORM to make balance transfer. 
you can implement additional function for the transfer in utils.py